lib/features/reps_core/reps_core.views.inc

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/**
+ * @file
+ * Code for the REPS core feature.
+ */
+
+/**
+ * Implements hook_views_query_alter().
+ */
+ function reps_core_views_query_alter(&$view, &$query) {
+
+  if ($view->name == 'reps_bean_blocks_global' && $view->current_display == 'block_reps_bean') {
+    foreach ($query->where as $index => &$condition_group) {
```

Comment:
So you have to choices here. Because what you want to achieve is actually to alter $query variable. Problem is the way you did it you have some mixed content at the end 
`$condition_group['conditions'][$id + 1]['operator'] = 'IS NULL';
 $query->where[$index]['type'] = 'OR';`

Altering the query directly avoid that and you don't pass $condition_group by reference.